subprojects/model-core/src/main/java/org/gradle/api/reflect/TypeOf.java

```diff
@@ -234,6 +234,17 @@ public String getSimpleName() {
         return type.getDisplayName();
     }
 
+    /**
+     * Concrete class.
+     *
+     * @since 4.11
+     * 
+     * @return Underlying Java Class of this type.
+     */
+    public Class<T> getConcreteClass() {
```

Comment:
Given that this is a public API, we should clarify in the javadocs what this means for various categories of types. eg what happens if this type represents a class? a parameterized type? an array type? what _should_ happen in these cases?